prow/README.md

```diff
@@ -1,71 +1,159 @@
   `/ok-to-test` for PRs authored by those not in the github org.
 
-See the [Prow command help](https://prow.apps.ci.metal3.io/command-help) for
+See the [Prow command help](https://prow.apps.test.metal3.io/command-help) for
 more information about who can run each prow command.
 
-## Prow Setup
+## GCS bucket
+
+Google Cloud Storage (GCS) is required to store job artifacts. Follow the docs to set
+up GCS and create a bucket with enough permissions.
 
-For more info about the CI cluster, see
-[project-infra/clusters/ci/README.md](../clusters/ci/README.md).
+1. Create a [bucket](https://cloud.google.com/storage/docs/creating-buckets)
+1. Create a [service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts)
+1. On the created bucket, grant `storage.objects.create` access for the service account and `storage.objectViewer`
+   permission to `allUser`.
+
+## Setup
 
 Prow was set up by following these instructions: https://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md
 
-```
-go get -u k8s.io/test-infra/prow/cmd/tackle
-tackle
-```
+1. Create DNS record(s) to point to your cluster.
+
+1. Prow components are running on `default` namespace, while `test-pods` namespace
+   is used by pods running actual CI jobs.
+
+1. [ghProxy](https://github.com/kubernetes/test-infra/tree/master/ghproxy) is a reverse proxy,
+   and is meant to reduce API token usage. We need persistent volume for ghProxy, and for that
+   purpose we rely on `hostPath` type of `PersistentVolume` in our cluster due to the limitation
+   in the underlying infrastructure. Ideally, you should use a dynamic storage provisioner.
+
+   Create a /mnt/prow directory to store the persistent volume data:
+
+     ```shell
+     $ sudo mkdir /mnt/prow
+     ```
+
+     Create the PersistentVolume:
+
+     ```shell
+     $ kubectl apply -f persistent_volume.yaml
+     ```
+
+1. Grant a user `cluster-admin` role in all namespaces to create cluster resources.
+
+    ```shell
+    $ kubectl create clusterrolebinding cluster-admin-binding-"${USER}" \
+      --clusterrole=cluster-admin --user="${USER}"
+    ```
+
+1. Create hmac token to give to GitHub for webhook validation and generate a secret out of it.
+
+    ```shell
+    $ openssl rand -hex 20 > /path/hmac-token
+    $ kubectl create secret generic hmac-token --from-file=hmac=/path/hmac-token
+    ```
+
+1. Create a personal access token for the GitHub
+   bot account. This should be done from [metal3-io-bot](https://github.com/metal3-io-bot)
+   GitHub bot account. You can follow this [link](https://github.com/settings/tokens)
+   to create the token. When generating the token, ensure to have the following scopes checked in.
+
+   - `public_repo` and `repo:status`
+   - `repo scope` for private repos
+   - `admin:org_hook` for a github org
+
+    Create a secret out of that token:
+    ```shell
+    $ kubectl create secret generic github-token --from-file=token=/path/github-token
```

Comment:
does it have to be in any specific namespace ?

Reply:
same as below, no